### PR TITLE
Verifier client

### PR DESCRIPTION
**Pull request by MadsChris92**
In order to not break integration tests this should only be merged when the following AVX PR is merged:
https://github.com/aion-dk/avx/pull/397

### CODE REVIEW

File: lib/av_client.ts [hunk 1]
```diff
@@ -346,7 +347,7 @@ export class AVClient implements IAVClient {
    * ```
    * @throws {@link NetworkError | NetworkError } if any request failed to get a response
    */
-    public async castBallot(_affidavit?: Affidavit): Promise<string> {
+    public async castBallot(_affidavit?: Affidavit): Promise<BallotBoxReceipt> {
```
Comment (sune-aion) [inline lib/av_client.ts:350]: I suppose this should return a `promise<string>` ? My notes says this should be the ballot tracking code. I think the BallotBoxReceipt type is deprecated.

File: lib/av_client.ts [hunk 2]
```diff
@@ -391,15 +392,20 @@ export class AVClient implements IAVClient {
...
    * @throws {@link NetworkError | NetworkError } if any request failed to get a response
    */
-  public async spoilBallotCryptograms(): Promise<void> {
-    // TODO: encrypt the vote cryptograms one more time with a key derived from `this.generateTestCode`.
-    //  A key is derived like: key = hash(test code, ballot id, cryptogram index)
-    // TODO: compute commitment openings of the voter commitment
-    // TODO: call the bulletin board to spoil the cryptograms. Send the encrypted vote cryptograms and voter commitment
-    //  opening. Response contains server commitment openings.
-    // TODO: verify the server commitment openings against server commitment and server empty cryptograms
+  public async spoilBallot(): Promise<void> {
```
Comment (sune-aion) [inline lib/av_client.ts:395]: Should return `promise<string>` ?

**Review by sune-aion — APPROVED** [review]

File: lib/av_client.ts (hunk 2 above)
Comment (MadsChris92) [reply lib/av_client.ts]: You're right. Fixed

File: lib/av_client.ts (hunk 1 above)
Comment (MadsChris92) [reply lib/av_client.ts:350]: In this particular case I think it should return an actual receipt. The ballot tracking code is something we use to track ballotCryptograms before casting them, unless i'm misunderstanding something. 

That being said, I dont think we have landed on exactly how a receipt should look like.

**Pull request merged**